Description: vmdk: Fix next_cluster_sector for compressed write (rounding)
Author: Radoslav Gerganov <rgerganov@vmware.com>
Origin: https://github.com/qemu/qemu/commit/3efffc3292d94271a15b1606b4a56adf6c6f04ed
Bug: https://bugs.launchpad.net/ubuntu/+source/qemu-kvm/+bug/1006655
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/qemu-kvm/+bug/1006655
Last-Update: 20150917

--- a/block/vmdk.c
+++ b/block/vmdk.c
@@ -1327,8 +1327,12 @@
 
     write_end_sector = DIV_ROUND_UP(write_offset + write_len, BDRV_SECTOR_SIZE);
 
-    extent->next_cluster_sector = MAX(extent->next_cluster_sector,
-                                      write_end_sector);
+    if (extent->compressed) {
+        extent->next_cluster_sector = write_end_sector;
+    } else {
+        extent->next_cluster_sector = MAX(extent->next_cluster_sector,
+                                          write_end_sector);
+    }
 
     if (ret != write_len) {
         ret = ret < 0 ? ret : -EIO;
